packages/transport/src/modules/sync/SynchronizedCollection.ts

```diff
@@ -87,6 +88,16 @@ export class SynchronizedCollection implements IDatabaseCollection {
 
         const newObjectJson = newObject.toJSON();
 
+        if (this.debugMode) {
+            const oldDocUpdated = Serializable.fromUnknown(await this.parent.read(newObject.id.toString()));
+
+            const readDiff = jsonpatch.compare(oldDocUpdated.toJSON(), oldObject.toJSON());
+            if (readDiff.length > 0) {
+                // eslint-disable-next-line no-console
+                console.error(`Object has been changed since last read.\n${new Error().stack}\n${JSON.stringify(readDiff, null, 2)}`);
```

Comment:
I'd like to give **way more** context on what's wrong here.

Also maybe we can find out that we are in a test any only log then? B/c I think the debugMode flag is also set in the connector, we then don't want to have this kind of log there.

Reply:
from my research one possibility would be to use `typeof globalThis.it === 'function';` to see if you are in a testing environment.

Reply:
great idea. but we cannot access process.env in prod code to be browser compatible ..

Reply:
I just tested this `if (typeof process === "object" && process.env.CI === "true")` and it works in node and browser.

I just need to test if the bundling in the app has any problem with it